libdnf/module/modulemd/profile/ModuleProfile.hpp

```diff
@@ -1,12 +1,33 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * Licensed under the GNU Lesser General Public License Version 2.1
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
 #ifndef LIBDNF_MODULEPROFILE_HPP
 #define LIBDNF_MODULEPROFILE_HPP
 
 
-#include "Profile.hpp"
-
 #include <memory>
+
 #include <modulemd/modulemd.h>
 
+#include "Profile.hpp"
+
```

Comment:
I am not sure if class NullProfile is beneficial. If the class can be replaced by nullptr we can simplify it. Additionally the profileMaker could be as a part of ModulePackage or ModuleMetadata. But I think it can be changed in next PR

Reply:
See [Null object pattern](https://en.wikipedia.org/wiki/Null_object_pattern)